app/ts/background/background-startup.ts

```diff
@@ -58,11 +58,29 @@ async function migrateAddressInfoAndContacts() {
 	if (userAddressBookEntries === undefined) return
 	const updated: AddressBookEntries = userAddressBookEntries.map(convertOldActiveAddressToAddressBookEntry)
 	if (updated.length > 0) {
-		await updateUserAddressBookEntries((previousEntries) => getUniqueItemsByProperties(updated.concat(previousEntries), ['address']))
+		await updateUserAddressBookEntriesV2Old((previousEntries) => getUniqueItemsByProperties(updated.concat(previousEntries), ['address']))
 		await browserStorageLocalRemove(['userAddressBookEntries'])
 	}
 }
-migrateAddressInfoAndContacts()
+async function migrateAddressInfoAndContactsFromV2ToV3() {
+	const userAddressBookEntries = (await browserStorageLocalGet(['userAddressBookEntriesV2'])).userAddressBookEntriesV2
+	const convertOldActiveAddressToAddressBookEntry = (entry: AddressBookEntry): AddressBookEntry => {
+		if ('chainId' in entry && entry.chainId !== undefined) return entry
```

Comment:
`entry` is already an `AddressBookEntry` type which should already guarantee the property `chainId`
```suggestion
		if (entry.chainId !== undefined) return entry
```